extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -554,9 +595,11 @@ private FullResponseHolder submitRequest(
           response = httpClient.go(request, new FullResponseHandler(StandardCharsets.UTF_8), httpTimeout).get();
         }
         catch (Exception e) {
-          Throwables.propagateIfInstanceOf(e.getCause(), IOException.class);
-          Throwables.propagateIfInstanceOf(e.getCause(), ChannelException.class);
-          throw Throwables.propagate(e);
+          if (e.getCause() instanceof IOException || e.getCause() instanceof ChannelException) {
```

Comment:
Can this be more explicit? things that catch `Exception` are dangerous.

For example, catching `catch(IOException ioe){throw ioe;}` first should handle the IOE cases. Then if it is the `httpClient.go` call that could be the issue, things that would need the `IOException` unwrapped can be handled explicitly. Then `catch(Exception re){throw new RuntimeException(e, "Unhandled exception");}` is probably good enough to handle all the other items that might crop up. Also make sure nothing here can throw `InterruptedException` or add a special handler for it please.

Things that `catch(Exception e)` can cause issues :-(

Reply:
Added an explicit handler for IOException and ChannelException, added a handler for InterruptedException